__tests__/app.test.js

```diff
@@ -342,4 +342,57 @@ describe("GET /api/users", () => {
             })
         })
     })
+})
+
+describe("GET /api/users (queries)", () => {
+    test('200: accepts sort_by query; title', () => {
+        return request(app)
+        .get('/api/reviews?sort_by=title')
+        .expect(200)
+        .then(({body: {reviews}}) => {
+            expect(reviews).toBeSortedBy('title', { descending: true })
+        })
+    })
+    test('400: bad sort_by query', () => {
+        return request(app)
+        .get('/api/reviews?sort_by=hello')
+        .expect(400)
+        .then(({ body }) => {
+            expect(body.msg).toBe("Bad Request.")
+        })
+    })
+    test('200: accepts order query; asc', () => {
+        return request(app)
+        .get('/api/reviews?order=asc')
+        .expect(200)
+        .then(({body: {reviews}}) => {
+            expect(reviews).toBeSortedBy('created_at')
+        })
+    })
+    test('400: bad sort_by ', () => {
+        return request(app)
+        .get('/api/reviews?order=hello')
+        .expect(400)
+        .then(({ body }) => {
+            expect(body.msg).toBe("Bad Request.")
+        })
+    })
+    test('200: accepts category; dexterity', () => {
+        return request(app)
+        .get('/api/reviews?category=dexterity')
+        .expect(200)
+        .then(({body: {reviews}}) => {
+            reviews.forEach((review) => {
+                expect(review.category).toBe("dexterity")
+            })
+        })
+    })
+    test('400: bad category ', () => {
+        return request(app)
+        .get('/api/reviews?category=office')
+        .expect(400)
+        .then(({ body }) => {
+            expect(body.msg).toBe("Bad Request.")
+        })
+    })
```

Comment:
Can we add testing here for a valid category query, but where there are no responses? I.e. category=children's+games (200).
order and sort_by incorrect requests should absolutely be a 400 like you've done because we want to control what options they have available with them based on the column name for sort_by and for order it has to be ASC or DESC, however with category we may want to think about it more carefully. If the category doesn't exist we are referring to a value of a column not existing, not a column name. So if it's not in there should that be a 400, or something different? Bit of a strange one I appreciate that so please drop me a message if it isn't clear.